neutron_plugin/rbac_policy.py

```diff
@@ -0,0 +1,198 @@
+#########
+# Copyright (c) 2018 GigaSpaces Technologies Ltd. All rights reserved
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+#  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  * See the License for the specific language governing permissions and
+#  * limitations under the License.
+
+from cloudify import ctx
+from cloudify.decorators import operation
+from cloudify.exceptions import NonRecoverableError
+
+from openstack_plugin_common import (
+    with_neutron_client,
+    use_external_resource,
+    validate_resource,
+    create_object_dict,
+    add_list_to_runtime_properties,
+    set_neutron_runtime_properties,
+    get_relationships_by_relationship_type,
+    delete_resource_and_runtime_properties,
+    COMMON_RUNTIME_PROPERTIES_KEYS,
+    OPENSTACK_ID_PROPERTY,
+    OPENSTACK_TYPE_PROPERTY
+)
+
+RBAC_POLICY_OPENSTACK_TYPE = 'rbac_policy'
+RBAC_POLICY_APPLIED_FOR_RELATIONSHIP_TYPE = \
+    'cloudify.openstack.rbac_policy_applied_to'
+
+
+def find_resource_to_apply_rbac_policy(ctx):
+    found_relationships = get_relationships_by_relationship_type(
+        ctx,
+        RBAC_POLICY_APPLIED_FOR_RELATIONSHIP_TYPE
+    )
+
+    if len(found_relationships) == 0:
+        ctx.logger.info(
+            'Resource for which RBAC policy may be applied '
+            'not found using {} relationship'
+            .format(RBAC_POLICY_APPLIED_FOR_RELATIONSHIP_TYPE)
+        )
+
+        return {}
+
+    if len(found_relationships) > 1:
+        raise NonRecoverableError(
+            'Multiple ({0}) resources for which RBAC policy may be applied '
+            'found using relationship {1}'
+            .format(
+                len(found_relationships),
+                RBAC_POLICY_APPLIED_FOR_RELATIONSHIP_TYPE
+            )
+        )
+
+    found_resource = found_relationships[0].target.instance
+    ctx.logger.info(
+        '{0} resource for which RBAC policy may be applied '
+        'found using {1} relationship)'
+        .format(found_resource, RBAC_POLICY_APPLIED_FOR_RELATIONSHIP_TYPE)
+    )
+
+    id = found_resource.runtime_properties.get(OPENSTACK_ID_PROPERTY, None)
+    type = found_resource.runtime_properties.get(
+        OPENSTACK_TYPE_PROPERTY,
+        None
+    )
+
+    if not id or not type:
+        ctx.logger.warn(
+            'Found using relationship resource has not defined either '
+            '"id" or "type" runtime_property. Skipping.'
+        )
+
+        return {}
+
+    return {
+        'object_type': type,
+        'object_id': id
+    }
+
```

Comment:
Seems like it would be better to do:
```
resource_dict = {}
if id:
  resource_dict['id'] = id
if type:
  resource_dict['type'] = type
return resource_dict```

Reply:
fine